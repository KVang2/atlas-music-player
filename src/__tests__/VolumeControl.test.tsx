import { render, screen, fireEvent } from "@testing-library/react";
import VolumeControl from "../components/VolumeControls";
import { describe, test, expect, vi } from "vitest";

describe('VolumeControl Component', () => {

    // test rendering
    test('Renders volume slider', () => {
        render(<VolumeControl volume={50} onVolumeChange={() => {}} />);

        const slider = screen.getByRole('slider');
        expect(slider).toBeInTheDocument();
        expect(slider).toHaveValue('50');
    });

    test('min and max values', () => {
        render(<VolumeControl volume={50} onVolumeChange={() => {}} />);
    
        const slider = screen.getByRole('slider');
        expect(slider).toHaveAttribute('min', '0');
        expect(slider).toHaveAttribute('max', '100');
    });

    // Test volume values
    test("onVolumeChange when volume get adjusted", () => {
        const onVolumeChange = vi.fn();
        render(<VolumeControl volume={50} onVolumeChange={onVolumeChange} />);

            const slider = screen.getByRole("slider");

            fireEvent.change(slider, { target: { value: "25" } });
    
            expect(onVolumeChange).toHaveBeenCalledTimes(1);
            expect(onVolumeChange).toHaveBeenCalledWith(25);
    });
});

describe("VolumeControl Snapshot", () => {
    test("renders correctly", () => {
            const { container } = render(<VolumeControl volume={50} onVolumeChange={() => {}} />);
            expect(container).toMatchSnapshot();
        });
    });