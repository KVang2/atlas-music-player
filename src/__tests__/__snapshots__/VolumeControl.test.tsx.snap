// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl Snapshot > renders correctly 1`] = `
<div>
  <div
    class="flex justify-center gap-2 m-3 pt-4 w-full"
  >
    <svg
      class="lucide lucide-volume2 m-2"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
    <input
      class="mt-4 w-full h-2 rounded-lg cursor-pointer dark:bg-darkvolume"
      max="100"
      min="0"
      type="range"
      value="50"
    />
  </div>
</div>
`;
